src/components/actions/dev_tools/routes.ts

```diff
@@ -0,0 +1,264 @@
+      {
+        const assignment = assignmentId
+          ? await db.getSingleReviewAssignment(assignmentId)
+          : await selectRandomReviewAssignment(applicationId, sectionCodes)
+        if (!assignment.allowedSections) assignment.allowedSections = sectionCodes
+        if (sections)
+          assignment.allowedSections = assignment.allowedSections.filter((section: string) =>
+            sections.includes(section)
+          )
+        await db.assignReview(assignment)
+
+        triggerPayload.trigger = Trigger.OnReviewAssign
+        triggerPayload.table = 'review_assignment'
+        triggerPayload.record_id = assignment.id
+
+        actionsOutput = await processTrigger(triggerPayload)
+        finalApplicationData = await getApplicationData({
+          applicationId,
+          reviewAssignmentId: assignmentId,
+        })
+      }
+      break
+
+    case 'ON_REVIEW_CREATE':
+      // Get assignment id -- either passed in, or pick one at random (assigned and highest stage/level)
+      {
+        const assignment = assignmentId
+          ? await db.getSingleReviewAssignment(assignmentId)
+          : await selectRandomReviewAssignment(applicationId, sectionCodes, true)
+        // Create a review record and review_decision record
+        const { id: revId } = await db.createReview(assignment.id)
+        await db.createReviewDecision(revId)
+        // Create one review response
+
+        // TO-DO: Make some review responses and give them dummy values?
+
+        triggerPayload.trigger = Trigger.OnReviewCreate
+        triggerPayload.table = 'review'
+        triggerPayload.record_id = revId
+
+        actionsOutput = await processTrigger(triggerPayload)
+        finalApplicationData = await getApplicationData({ applicationId, reviewId: revId })
+      }
+      break
+
+    case 'ON_REVIEW_RESTART':
+      {
+        const revId = reviewId ?? (await getRandomReviewId(applicationId))
+
+        triggerPayload.trigger = Trigger.OnReviewRestart
+        triggerPayload.table = 'review'
+        triggerPayload.record_id = revId
+
+        actionsOutput = await processTrigger(triggerPayload)
+        finalApplicationData = await getApplicationData({ applicationId, reviewId: revId })
+      }
+      break
+
+    case 'ON_REVIEW_SUBMIT':
+      {
+        const revId = reviewId ?? (await getRandomReviewId(applicationId))
+        await db.updateReviewDecision(revId, decision ?? Decision.Conform, comment ?? 'No comment')
+
+        triggerPayload.trigger = Trigger.OnReviewSubmit
+        triggerPayload.table = 'review'
+        triggerPayload.record_id = revId
+
+        actionsOutput = await processTrigger(triggerPayload)
+        finalApplicationData = await getApplicationData({ applicationId, reviewId: revId })
+      }
+      break
+
+    case 'RESET':
+      await db.resetApplication(applicationId)
+      finalApplicationData = await getApplicationData({ applicationId })
+      break
```

Comment:
No clear what this one does : (

Reply:
![Screen Shot 2022-10-14 at 12 48 51 PM](https://user-images.githubusercontent.com/5456533/195732740-90745f24-c9a2-4f9d-91b8-04214c228983.png)

So you can reset and start again. Otherwise you'd only be able to go through it once per application/type. 
(And also serves as something of a garbage collector, since using this tool will leave a bunch of crap in the database)